src/js/test/rpc/server.test.ts

```diff
@@ -657,5 +605,108 @@ describe("Server", function () {
         });
       }
     );
+
+    it(
+      "should response [] in record on request_process_replay if" +
+        "there is a error on coprocessor script, and its policy error is " +
+        "SkipOnFailure",
+      () => {
+        const handle = createHandle();
+        // add unhandle expetion to coprocessor function
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+        // @ts-ignore
+        handle.coprocessor.apply = (record) => record.badAtt.ribute;
+        const repositoryMock = sinonInstance.stub(
+          Repository.prototype,
+          "getHandlesByCoprocessorIds"
+        );
+        repositoryMock.returns([handle]);
+
+        const request: ProcessBatchRequest = {
```

Comment:
Future work, lots of test code ends up being terse, maybe we can have some additional functions or a small harness that can setup the common mocked server and/or client and perform most of the other common code